src/Corax/IndexWriter.cs

```diff
@@ -1829,177 +1838,426 @@ private void InsertSpatialField(Tree entriesToSpatialTree, IndexedField indexedF
             }
         }
 
-        private void InsertTextualField(Tree entriesToTermsTree, IndexedField indexedField, Span<byte> tmpBuf, ref Slice[] sortedTermsBuffer)
+        
+        private class TextualFieldBuffers : IDisposable
```

Comment:
Maybe let's put it in dedicated file? IndexWriter is already pretty big class

Reply:
Let's leave it for now. I see that we have multiple structures embedded inside IndexWriter. Many of them could be moved to dedicated files. Let's do the cleanup after merging this PR.

Reply:
Yes.